### PR TITLE
Add os_recordset module

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

os_recordset
##### ANSIBLE VERSION

```
(venv) ubuntu@designate:~$ ansible --version
ansible 2.2.0 (devel fb7940fc50) last updated 2016/05/05 17:40:06 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 9db1233521) last updated 2016/05/05 17:40:18 (GMT +200)
  lib/ansible/modules/extras: (add_os_recordset 3027dc7f7a) last updated 2016/05/16 16:57:37 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

This module allows to manage OpenStack Designate recordsets.

```
(venv) ubuntu@designate:~$ cat test_os_recordset.yml

---
- hosts: localhost
  connection: local
  gather_facts: false
  tasks:
  - os_recordset:
      cloud: devstack-admin
      state: absent
      zone: example.net.
      name: www
      recordset_type: A
      records: ['10.1.1.1']
      description: Test recordset
      ttl: 3600

(venv) ubuntu@designate:~$ ansible-playbook -i hosts test_os_recordset.yml -vvv
No config file found; using defaults

PLAYBOOK: test_os_recordset.yml ************************************************
1 plays in test_os_recordset.yml

PLAY [localhost] ***************************************************************

TASK [os_recordset] ************************************************************
task path: /home/ubuntu/test_os_recordset.yml:6
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1463410507.58-31888303673298 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1463410507.58-31888303673298 `" )'
<localhost> PUT /tmp/tmpb37P_K TO /home/ubuntu/.ansible/tmp/ansible-tmp-1463410507.58-31888303673298/os_recordset
<localhost> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/env python /home/ubuntu/.ansible/tmp/ansible-tmp-1463410507.58-31888303673298/os_recordset; rm -rf "/home/ubuntu/.ansible/tmp/ansible-tmp-1463410507.58-31888303673298/" > /dev/null 2>&1'
changed: [localhost] => {"changed": true, "invocation": {"module_args": {"api_timeout": null, "auth": null, "auth_type": null, "availability_zone": null, "cacert": null, "cert": null, "cloud": "devstack-admin", "endpoint_type": "public", "key": null, "region_name": null, "timeout": 180, "verify": true, "wait": true}, "module_name": "os_recordset"}, "recordset": {"action": "CREATE", "created_at": "2016-05-16T14:55:08.000000", "description": "Test recordset", "id": "59dd2fb3-df03-4705-a54d-e294c29879f8", "links": {"self": "http://10.0.3.183:9001/v2/zones/9508e177-41d8-434e-962c-6fe6ca880af7/recordsets/59dd2fb3-df03-4705-a54d-e294c29879f8"}, "name": "www.example.net.", "records": ["10.1.1.1"], "status": "PENDING", "ttl": 7200, "type": "A", "updated_at": null, "version": 1, "zone_id": "9508e177-41d8-434e-962c-6fe6ca880af7"}}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @rcarrillocruz. This module is going into community review.

We notice that you are one of the maintainers (@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @rcarrillocruz for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by rcarrillocruz** [comment]
Rechecking tests, can't see how the failure is linked to my commit....

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_recordset.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                               ('state', 'present',
+                                ['recordset_type', 'records'])],
+                           supports_check_mode=True,
+                           **module_kwargs)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+    if StrictVersion(shade.__version__) <= StrictVersion('1.8.0'):
+        module.fail_json(msg="To utilize this module, the installed version of"
+                             "the shade library MUST be <=1.8.0")
```
Comment (timl) [inline cloud/openstack/os_recordset.py]: Should this read "the shade library MUST be >1.8.0"?

Comment (rcarrillocruz) [inline cloud/openstack/os_recordset.py]: Good eyes, it should. Thanks.

Comment (timl) [inline cloud/openstack/os_recordset.py]: Also, you're missing a space between `of` and `the`.  Sorry to nitpick..

**Comment by Shrews** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @rcarrillocruz for this PR, and thanks @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
Hmm is check_mode really implemented for state=absent? Please verify.

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Shrews** [comment]
Let's not pop() params because it affects verbose output.

needs_revision

**Comment by gregdek** [comment]
Thanks @rcarrillocruz for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**